pkg/parse/run.go

```diff
@@ -111,17 +118,33 @@ func Run(ctx context.Context, p Parser) {
 				continue
 			}
 			run(ctx, p, trigger, state)
-			retryTimer.Reset(opts.retryPeriod)
+
+			retryTimer.Reset(opts.retryPeriod)               // Schedule retry attempt
+			statusUpdateTimer.Reset(opts.statusUpdatePeriod) // Schedule status update attempt
+
+		// Update the sync status to report management conflicts (from the remediator).
+		case <-statusUpdateTimer.C:
+			// Skip sync status update if the .status.sync.commit is out of date.
```

Comment:
I like the idea of using another timer to make sure the status update and parse-apply-watch loop are mutually exclusive.